### PR TITLE
toggle class to nav on scroll

### PR DESCRIPTION
**Pull request by neu5**

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -1 +1,59 @@
...
+  	};
+  };
 
+  function getElementByClassName(className) {
+    if (document.getElementsByClassName) {
+      return document.getElementsByClassName(className)[0];
+    }
+
+    if (document.querySelectorAll) {
+      return document.querySelectorAll((' ' + className).replace(/ +/g, '.'))[0];
```
Comment (CezaryDanielNowak) [inline js/main.js]: IE8 supports querySelectorAll, I think, there is no need to polyfill

File: js/main.js [hunk 2]
```diff
@@ -1 +1,59 @@
...
+    }
+  };
+
+  function addClass(elem, className) {
+    var re = new RegExp(className, 'g');
+    var elemClassName = elem.className;
+
+    if (re.test(elemClassName) === false) {
```
Comment (CezaryDanielNowak) [inline js/main.js]: It's not really booletproof. Try 'foo' on 'foobar' - it will be true and class won't be added

File: js/main.js [hunk 3]
```diff
@@ -1 +1,59 @@
...
+
+  function removeClass(elem, className) {
+    var re = new RegExp(className, 'g');
+    var elemClassName = elem.className;
+
+    if (re.test(elemClassName) === true) {
+      elemClassName = elemClassName.replace(className, '');
+      // remove not need one space left after replacing `className`
+      elem.className = elemClassName.slice(0, elemClassName.length - 1);
+    }
```
Comment (CezaryDanielNowak) [inline js/main.js]: I'd find node.classList polyfill than writing custom function for add/remove class

File: js/main.js (hunk 1 above)
Comment (neu5) [inline js/main.js]: true, but `getElementsByClassName` it's not supported by IE8, that's why it's at the top :)

File: js/main.js (hunk 3 above)
Comment (neu5) [inline js/main.js]: so much code https://github.com/eligrey/classList.js/blob/master/classList.js :D

File: js/main.js (hunk 2 above)
Comment (neu5) [inline js/main.js]: better? :P

File: js/main.js [hunk 4]
```diff
@@ -1 +1,59 @@
+(function () {
+  function debounce(func, wait, immediate) {
+  	var timeout;
```
Comment (adambabik) [inline js/main.js]: Uninitialized variable.

File: js/main.js [hunk 5]
```diff
@@ -1 +1,59 @@
...
+  	return function() {
```
Comment (adambabik) [inline js/main.js]: Unnamed function. `function debounceInner()` or sth.

File: js/main.js [hunk 6]
```diff
@@ -1 +1,59 @@
...
+  		var context = this, args = arguments;
+  		var later = function() {
```
Comment (adambabik) [inline js/main.js]: Unnamed fn.

File: js/main.js [hunk 7]
```diff
@@ -1 +1,59 @@
+(function () {
+  function debounce(func, wait, immediate) {
+  	var timeout;
+  	return function() {
+  		var context = this, args = arguments;
```
Comment (adambabik) [inline js/main.js]: `arguments` should be converted to `Array`.

File: js/main.js (hunk 1 above)
Comment (adambabik) [inline js/main.js]: Why not `querySelector`? Only the first item is returned anyway.

File: js/main.js [hunk 8]
```diff
@@ -1 +1,59 @@
...
+    if (document.querySelectorAll) {
+      return document.querySelectorAll((' ' + className).replace(/ +/g, '.'))[0];
+    }
+  };
+
+  function addClass(elem, className) {
+    var re = new RegExp(className + '$', 'g');
+    var elemClassName = elem.className;
+
+    if (re.test(elemClassName) === false) {
```
Comment (adambabik) [inline js/main.js]: ??? `!re.test(elemClassName)`.

File: js/main.js [hunk 9]
```diff
@@ -1 +1,59 @@
...
+      return document.getElementsByClassName(className)[0];
+    }
+
+    if (document.querySelectorAll) {
+      return document.querySelectorAll((' ' + className).replace(/ +/g, '.'))[0];
+    }
+  };
+
+  function addClass(elem, className) {
+    var re = new RegExp(className + '$', 'g');
```
Comment (adambabik) [inline js/main.js]: `indexOf()` should be enough as a substring test.

File: js/main.js [hunk 10]
```diff
@@ -1 +1,59 @@
...
+    var elemClassName = elem.className;
+
+    if (re.test(elemClassName) === false) {
+      elem.className = elemClassName + ' ' + className;
+    }
+  }
+
+  function removeClass(elem, className) {
```
Comment (adambabik) [inline js/main.js]: `elem.className = elem.className.replace(re, '')` should be enough.

Comment (adambabik) [inline js/main.js]: You can end up with multiple spaces inside the string so maybe make regex which handles that.

File: js/main.js [hunk 11]
```diff
@@ -1 +1,59 @@
...
+      elem.className = elemClassName.slice(0, elemClassName.length - 1);
+    }
+  }
+
+  var SCROLL_TOP_LIMIT = 25;
+  var navMain = getElementByClassName('nav-main');
+
+  var scrollHandler = debounce(function() {
+    if (window.scrollY > SCROLL_TOP_LIMIT) {
+      addClass.call(this, navMain, 'nav-small')
```
Comment (adambabik) [inline js/main.js]: What is `this` here? What do you need it for?

File: js/main.js (hunk 8 above)
Comment (neu5) [inline js/main.js]: it's less readable :P

File: js/main.js (hunk 11 above)
Comment (neu5) [inline js/main.js]: nothing important, can be `null` :)

File: js/main.js (hunk 1 above)
Comment (neu5) [inline js/main.js]: I'll use `classList` as @CezaryDanielNowak suggested anyway

File: js/main.js (hunk 4 above)
Comment (neu5) [inline js/main.js]: you can talk to [David Walsh](https://davidwalsh.name/javascript-debounce-function) :yum:

File: js/main.js (hunk 5 above)
Comment (neu5) [inline js/main.js]: I'll use `arrow function` when I'll add `core-js`

File: js/main.js (hunk 9 above)
Comment (neu5) [inline js/main.js]: I'll use `classList`

File: js/main.js (hunk 10 above)
Comment (neu5) [inline js/main.js]: I'll use `classList`

File: js/main.js (hunk 11 above)
Comment (neu5) [inline js/main.js]: I'll use `classList`

File: js/main.js (hunk 8 above)
Comment (neu5) [inline js/main.js]: I'll use `classList`

**Comment by neu5** [comment]
`classList` polyfill needs to be added

**Pull request merged**